app/src/main/java/com/example/wallto/base/Logger.kt

```diff
@@ -0,0 +1,41 @@
+/*
+ * Created by Mark Abramenko on 10/30/19 8:57 PM
+ * Copyright (c) 2019 . All rights reserved.
+ * Last modified 10/30/19 8:57 PM
+ */
+
+package com.example.wallto.base
+
+import android.util.Log
+// пример логгера который можно дергать из любой части кода, даже из домена, который не связан с
+// андроид фреймворком
+object Logger {
+    private var writer: LogWriter = LogWriterConsole()
+
+    fun setWriter(writer: LogWriter) {
```

Comment:
вот этот метод делает синглтон с состоянием. Это можно исправить добавив флаг инициализации или убрав дефолтное значение и сделав переменную lateinit. После чего проверять ее состояние и принимать нового писателя только если переменная еще не была инициализированна
```suggestion
private lateinit var writer: LogWriter
    fun initWriter(writer: LogWriter) {
        if (!::this.writer.isInitialized) {
            this.writer = writer
        }
    }
```